### DISCUSSION TITLE
Automatically add slash when adding Month/Year

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5790] [Feature Requests]
**Requested By:** RBGE

>Request is for the / to be automatically added when adding a Month/Year, like with the Full Date, rather than the user needing to add the /.

**Comment by maxpatiiuk** [comment]
@grantfitzsimmons is this about the partial date picker?

The format of the partial date picker is influenced by the browser

<img width="445" alt="Screenshot 2023-04-24 at 15 55 06" src="https://user-images.githubusercontent.com/40512816/234114840-0d83bd17-fa87-421a-9aac-144bbf89e685.png">

That is, unless they disabled the accessible date picker, in which case I would suggest they re-enable it.

**Comment by grantfitzsimmons** [comment]
They may have disabled it to make the date format consistent on all machines (not uncommon). If it is disabled, can we make their request happen?

**Comment by maxpatiiuk** [comment]
when implementing this, have to make sure not to make it frustrating for the user.

Example:
- When `/` is automatically inserted, the cursor should be moved to be after `/`
- When pasting in text, `/` should not be automatically inserted to not corrupt inserted text (as it's better to assume it's already formatted or would be formatted by the user)
- When the cursor is after `/` and the user hits backspace, have to make sure not to add `/` right back.
- When the cursor is after `/` and the user hits backspace, does backspace erase just `/` or also one character before it? (i.e, since `/` wasn't inserted by them, by hitting backspace they might mean to erase the last typed character, not the last character in the box)

All doable, but not trivial to get right.

Perharps instead of automatically inserting `/`, we can split the date field into multiple? In that case, this feature request can be combined with https://github.com/specify/specify7/issues/2137.

In that case, instead of all of the above problems there are two possible solutions:
- Complicated: if first box accepts 4 characters and they typed in 4 characters, move focus to the next box right away. if in the next box and they hit backspace to erase the last character, move focus back to previous box. If in the next box, with no value, and they hit backspace, move focus to previous box and erase the last character from the previous box.
- Simple: don't do any focus management, let the user do it all
